sote/game/scenes/world-gen.lua

```diff
@@ -44,10 +80,14 @@ end
 local function cache_tile_coord()
 	print("Caching tile coordinates...")
 
-	for _, tile in pairs(WORLD.tiles) do
-		local lat, lon = tile:latlon()
-		local q, r, face = hex.latlon_to_hex_coords(lat, lon - math.pi, wg.world.size) -- latlon_to_hex_coords expects lon in range [-pi, pi]
-		wg.world:cache_tile_coord(tile.tile_id, q, r, face)
+	-- it's faster to load the pre-calculated coordinates from a file than to calculate them on the fly
+	for row in fu.csv_rows("sote\\libsote\\data\\hex_mapping.csv") do
```

Comment:
This basically hardcodes world size. That's fine as an optimization but if we are porting world gen, it has world size in the api. Can we have a check for world size (needed in a world gen config) before using a cached version?

Reply:
This world-gen scene is just a playground for testing the port, it's not meant to become permanent. I introduced the hard-coded caching to save a bit of time when testing/debugging. It's just a mapping between the square tiles of the world and the hex coordinates of the ported one, so I can move data between the two.
The world allocator and the api still take world size as a parameter and I've used it quite often with a small value while debugging the Goldberg topology, for example.
I would leave it like that for now.